File: test/leetcode/tree/binary-tree-right-side-view.test.ts
import { array2tree } from '../../src/tree/common/tree-node';
import { rightSideView } from '../../src/tree/binary-tree-right-side-view';

describe('binary tree right side view', () => {
  test('binary tree right side view', async () => {
    const root = array2tree([1, 2, 3, null, 5, null, 4]);

    expect(rightSideView(root)).toStrictEqual([1, 3, 4]);
  });
});

File: test/leetcode/tree/binary_tree_right_side_view_test.py
from leetcode.tree.binary_tree_right_side_view import Solution
from leetcode.tree.common.tree_node import array2tree


soln = Solution()


def test_case_1():
    root = array2tree([1, 2, 3, None, 5, None, 4])

    assert soln.rightSideView(root) == [1, 3, 4]
